ramda.js

```diff
@@ -4705,6 +4707,42 @@
     R.gte = op(function _gte(a, b) { return a >= b; });
 
 
+    function createMaxMin(comparator, initialVal) {
+        return function(list) {
+            if (arguments.length === 0) {
+                throw noArgsException();
+            }
+            var idx = -1, winner = initialVal, computed;
+            while (++idx < list.length) {
+                computed = +list[idx];
+                if (comparator(computed, winner)) {
+                    winner = computed;
+                }
+            }
+            return winner;
+        };
+    }
+
+    function createMaxMinWith(comparator) {
+        return function(valueComputer, list) {
+            if (!(list && list.length > 0)) {
+                return;
```

Comment:
Right,  and it's still a reasonable question.  Ramda's focus on values rather than references means that this coercion is reasonable.  But it doesn't particularly imply that it's required.

The main reason that I really do prefer coercion in this case is that if don't do it, then we lose the guarantee that

    max([a, b, ...]) == max([b, a, ...])

and that seems too much to give up.

`maxBy`/`minBy` are very different sorts of beasts.  They really do feel closer to `find`, and the use of references isn't so disconcerting.  Even here, I'm tempted to return a clone of the value, but that would simply be to prove a point; I've seen no practical reason for it.

And here you never could offer such an invariant as the above.